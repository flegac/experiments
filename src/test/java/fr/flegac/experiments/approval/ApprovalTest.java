package fr.flegac.experiments.approval;

import org.approvaltests.legacycode.LegacyApprovals;
import org.approvaltests.reporters.JunitReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;

@UseReporter(JunitReporter.class)
public class ApprovalTest {

    final static String[] toto = {
        "a", ""
    };

    final static String[] tata = {
        "b", ""
    };

    final static String[] titi = {
        "", "c"
    };

    @Test
    public void _test1() throws Exception {
        LegacyApprovals.LockDown(this, "test1", toto, tata, titi);
    }

    @Test
    public void _test2() throws Exception {
        LegacyApprovals.LockDown(this, "test2", toto, tata, titi);
    }

    @Test
    public void _test3() throws Exception {
        LegacyApprovals.LockDown(this, "test3", toto, tata, titi);
    }

    public String test1(final String toto, final String tata, final String titi) {
        final Bidon item = new Bidon(toto, tata, titi);
        item.doSomething();
        return item.toString();
    }

    public String test2(final String toto, final String tata, final String titi) {
        final Bidon item = new Bidon(toto, tata, titi);
        item.setTata(titi);
        item.setToto(tata);
        item.setTiti(toto);

        return item.toString();
    }

    public String test3(final String toto, final String tata, final String titi) {
        final Bidon item = new Bidon(toto, tata, titi);

        return item.getTata() + item.getToto() + item.getTiti();
    }

}
