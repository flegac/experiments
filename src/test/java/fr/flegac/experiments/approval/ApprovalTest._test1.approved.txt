[a, b, ] = Bidon [toto=ab, tata=b, titi=] 
[, b, ] = Bidon [toto=b, tata=b, titi=] 
[a, , ] = Bidon [toto=a, tata=, titi=] 
[, , ] = Bidon [toto=, tata=, titi=] 
[a, b, c] = Bidon [toto=a, tata=b, titi=c] 
[, b, c] = Bidon [toto=, tata=b, titi=c] 
[a, , c] = Bidon [toto=ca, tata=, titi=c] 
[, , c] = Bidon [toto=c, tata=, titi=c] 
